Add promo codes with department selection

Requiem4soul opened the pull request:
**Добавление промокодов с выбором отдела (в будущем необходимо сделать слияние с promo)**
1) Добавлена возможность выбирать нужный отдел. В случае если промокод был написан правильно, активен, то к выбранному отделу добавляется время на конкретные роли из этого отдела
2) При активации промокода, выпадающий список видит только активировавший команду пользователь. Это поможет защитить от "рофлов", когда другие игроки за пользователя будут выбирать ему отдел
3) В alembic/versions было добавлено создание 2-ух таблиц для данного кога. Совместимо с upgrade/downgrade 
4) Переработана логика usages для таблицы "choosing_promo". Теперь usages по умолчанию начинается с нуля, и после каждого использования увеличивается на один. Возможная проблема - невозможно ограничить количество использований промокодов

![изображение](https://github.com/user-attachments/assets/5b83db8a-e6c5-40a0-a16f-67aa88fcb08a)

Requiem4soul committed:
Создание кога на ютуб промо (доделать полный перенос в промо)

1) Был создан отдельный ког для промокода с выбором отдела и начислением времени на конкретный отдел

Task future:
Необходимо перенести целиком это в промо

Requiem4soul committed:
ХотФикс с проверкой времени

1) Теперь, если промокод истёк, то в ошибке будет это указано. Раньше было "Такого промокода не существует"

Requiem4soul committed:
ХотФикс

1) Исправлено несоответствие название отделов

pguin-sudo commented on orienteer/bot/__main__.py:
Глянуть, почему тут CommandInteraction

pguin-sudo commented on orienteer/bot/calls/info.py:
Вот это странная шука

pguin-sudo commented on orienteer/bot/cogs/ytpromo.py:
Вынести DropDownView в bot/utils

pguin-sudo commented on orienteer/bot/utils/ytpromo.py:
Перенести в bot/utils/content_locale.py  и bot/utils/embeds.py

pguin-sudo commented on orienteer/general/data/orienteer/models/transactions.py:
Глянуть зачем и убрать это

pguin-sudo commented on orienteer/general/data/orienteer/repositories/promo.py:
Перенести из promo.py в ytpromo.py

pguin-sudo reviewed (changes requested):
Надо доделать чутка :)

Requiem4soul committed:
Fixes everything except DropDownView

All that's left is to get DropDownView into the utilities

Requiem4soul committed:
Fix DropDownView

1) The classes related to the displayed interfacing have been moved to utills/ytpromo.py. Tests were performed, no errors were detected
